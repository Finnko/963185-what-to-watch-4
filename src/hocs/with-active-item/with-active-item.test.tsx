import * as React from 'react';
import * as renderer from 'react-test-renderer';
import withActiveItem from './with-active-item';

const MockComponent = () => {
  return (
    <ul>
      <li>Item 1</li>
      <li>Item 2</li>
      <li>Item 3</li>
    </ul>
  );
};

const MockComponentWrapped = withActiveItem(MockComponent);


describe(`withActiveItem works correctly`, () => {
  it(`withActiveItem is rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        activeItem={`Item 1`}
        onActiveItemChange={() => {}}
      />
    ), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
