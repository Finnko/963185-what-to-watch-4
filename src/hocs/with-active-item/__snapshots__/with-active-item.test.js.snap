// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveItem works correctly withActiveItem is rendered correctly 1`] = `
<ul>
  <li>
    Item 1
  </li>
  <li>
    Item 2
  </li>
  <li>
    Item 3
  </li>
</ul>
`;
